wdl-grammar/src/v1/lint/document_preamble.rs

```diff
@@ -0,0 +1,272 @@
+//! Incorrect document preamble
+
+use std::collections::VecDeque;
+
+use nonempty::NonEmpty;
+use pest::iterators::Pair;
+use wdl_core::concern::code;
+use wdl_core::concern::lint;
+use wdl_core::concern::lint::Rule;
+use wdl_core::concern::lint::Tag;
+use wdl_core::concern::lint::TagSet;
+use wdl_core::concern::Code;
+use wdl_core::file::Location;
+use wdl_core::Version;
+
+use crate::v1;
+
+/// Detects improper document preamble.
+/// Checks for whitespace, comments, and version declaration placement.
+#[derive(Debug)]
+pub struct DocumentPreamble;
+
+impl<'a> DocumentPreamble {
+    /// Generates a lint warning for an improperly placed version
+    /// declaration.
+    fn misplaced_version(&self, location: Location) -> lint::Warning
+    where
+        Self: Rule<&'a Pair<'a, v1::Rule>>,
+    {
+        // SAFETY: this error is written so that it will always unwrap.
+        lint::warning::Builder::default()
+            .code(self.code())
+            .level(lint::Level::Low)
+            .tags(self.tags())
+            .subject("Improperly placed version declaration")
+            .body(
+                "The version declaration must be the first line of a WDL document or one blank \
+                 line after header comments.",
+            )
+            .push_location(location)
+            .fix(
+                "Move the version declaration to the first line of the WDL document or one blank \
+                 line after header comments.",
+            )
+            .try_build()
+            .unwrap()
+    }
+
+    /// Generates a lint warning for leading whitespace.
+    fn leading_whitespace(&self, location: Location) -> lint::Warning
+    where
+        Self: Rule<&'a Pair<'a, v1::Rule>>,
+    {
+        // SAFETY: this error is written so that it will always unwrap.
+        lint::warning::Builder::default()
+            .code(self.code())
+            .level(lint::Level::Low)
+            .tags(self.tags())
+            .subject("Leading whitespace detected")
+            .body("No whitespace is allowed at the beginning of the document")
+            .push_location(location)
+            .fix("Remove leading whitespace.")
+            .try_build()
+            .unwrap()
+    }
+}
+
+impl<'a> Rule<&Pair<'a, v1::Rule>> for DocumentPreamble {
+    fn code(&self) -> Code {
+        // SAFETY: this manually crafted to unwrap successfully every time.
+        Code::try_new(code::Kind::Error, Version::V1, 9).unwrap()
+    }
+
+    fn tags(&self) -> TagSet {
+        TagSet::new(&[Tag::Spacing])
```

Comment:
```suggestion
        TagSet::new(&[Tag::Spacing, Tag::Style])
```
Maybe pedantic, but I think it's better to be explicit here that we want this tagged with `Style` even though that's implicitly added with `Spacing`.